test/deprecated/dotnl.skipif

```diff
@@ -1 +1 @@
-CHPL_REGEXP != re2
+CHPL_RE2==none
```

Comment:
This is an example of a test that has changed behavior

Reply:
> I ask because there were a number of test specific files that went from checking if CHPL_REGEXP != re2 to checking CHPL_RE2==none. These test specific files are used outside of a printchplenv context so would not necessarily see the "default" setting that is handled within the Chapel program itself.

I haven't been following this PR closely, but wanted to note that skipifs and most other files have the printchplenv injected into them. I think the only files we don't do that for today are precomp/prediff/preexec, but I'm not positive about that.

I'm not sure if there's a reason the original skipifs were written in the old style, and I don't really see a reason to change it, but at least for this particular one I don't think it'll do any harm.

Reply:
We wanted to move away from having RE2 in the value (only in the name), which is why the `!= re2` and `== re2` `skipif` tests were changed to an equivalent test with `none`. `none` is a pretty universal value to compare against, and when the value is not `none`, we consider that it is set to some other value which means that the RE2 feature is "enabled" and so it can be tested.

Yes, `printchapelenv` is used to get the `skipif` values, so the `chpl_re2.py` Python module is used to infer the value for `CHPL_RE2`. It is either inferred as `bundled` or `none`. It is not inferred as unset, nor was the `CHPL_REGEXP` variable inferred as unset in the past. If the environment, command-line and `chplopts` do not set the variable, then the `chpl_re2.py` script infers a value based on the success or failure of the top-level `Makefile` in building `RE2`. We are preserving the behavior as shown in the table above.

@lydia-duncan: Are you saying that particular tests have changed behavior because of this PR's changes, and that it is the wrong behavior? I would like specifics, because the `skipif` tests should not have changed the behavior. The same tests are skipped under the same conditions as before `CHPL_RE2=(unset, none, bundled) <=> CHPL_REGEXP=(unset, none, re2)`.

Are you saying that the behavior of particular tests changed for other reasons?

Are you questioning why some of the tests had their `.good` files changed because of the renaming of `CHPL_REGEXP` to `CHPL_RE2` and the new values of `bundled` and `none` instead of `re2` and `none`?

Are you saying that the switch from `CHPL_REGEXP` to `CHPL_RE2` has changed the behavior of tests in unexpected ways?

Reply:
> We wanted to move away from having RE2 in the value (only in the name), which is why the `!= re2` and `== re2` `skipif` tests were changed to an equivalent test with `none`. `none` is a pretty universal value to compare against, and when the value is not `none`, we consider that it is set to some other value which means that the RE2 feature is "enabled" and so it can be tested.
> 
> Yes, `printchapelenv` is used to get the `skipif` values, so the `chpl_re2.py` Python module is used to infer the value for `CHPL_RE2`. It is either inferred as `bundled` or `none`. It is not inferred as unset, nor was the `CHPL_REGEXP` variable inferred as unset in the past. If the environment, command-line and `chplopts` do not set the variable, then the `chpl_re2.py` script infers a value based on the success or failure of the top-level `Makefile` in building `RE2`. We are preserving the behavior as shown in the table above.
> 
> @lydia-duncan: Are you saying that particular tests have changed behavior because of this PR's changes, and that it is the wrong behavior? I would like specifics, because the `skipif` tests should not have changed the behavior. The same tests are skipped under the same conditions as before `CHPL_RE2=(unset, none, bundled) <=> CHPL_REGEXP=(unset, none, re2)`.

I was concerned because I have encountered cases in the past where comparing a CHPL_ environment variable against 'none' when the environment variable was not explicitly set caused cases to be missed.  I wasn't sure which test helper files were impacted by it or if I was remembering correctly, which is why I tagged Elliot.  I trust his memory on this, so I think it is fair for you to only look at precomp/prediff/preexec.

> 
> Are you saying that the behavior of particular tests changed for other reasons?

No.  I am only concerned about cases when `CHPL_REGEXP == <something>` was transformed to `CHPL_RE2 != <something else>` or `CHPL_REGEXP != <something>` was transformed to `CHPL_RE2 == <something else>`.

> 
> Are you questioning why some of the tests had their `.good` files changed because of the renaming of `CHPL_REGEXP` to `CHPL_RE2` and the new values of `bundled` and `none` instead of `re2` and `none`?

Definitely not

> 
> Are you saying that the switch from `CHPL_REGEXP` to `CHPL_RE2` has changed the behavior of tests in unexpected ways?

Nope, I'm solely concerned about cases where the files themselves are changed in the way I have described.

Reply:
I did not see any usage of `CHPL_REGEXP` or `CHPL_RE2` in `precomp`, `preexec` or `prediff`. It's only in the `skipif` files, which definitely call `printchplenv` and thus `chpl_re2.py`.

Reply:
Thanks.  I'm still not certain why you changed the other conditions to the opposite style of check but I'm satisfied that it doesn't make a material different.